### PR TITLE
Config UI write correct model id on save

### PR DESCRIPTION
**Pull request by krishnaglick**
## Change description

> Model ID's on Source configs were not friendly.

## Related issues

Does this PR fix a Github Issue?

### Development

- [X] Application changes have been tested appropriately

### Impact

- [X] Code follows company security practices and guidelines
- [X] Security impact of change has been considered
- [X] Performance impact of change has been considered
- [X] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

>

### Code review

- [X] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [X] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Review by evantahler — CHANGES_REQUESTED** [review]
This is a good fix... but it implies that Group, GrouparooRecord, and Destination also likely have a similar problem.  Can you please check those too?

**Comment by krishnaglick** [comment]
> This is a good fix... but it implies that Group, GrouparooRecord, and Destination also likely have a similar problem. Can you please check those too?
Will fix

**Comment by pedroslopez** [comment]
@krishnaglick note that this only happens when the model you're assigning has been created in the same `roo config` session. If you exit out and start creating things with an existing model, you won't see it. I was trying out some things working on another story and did see the issue with groups as well

File: core/src/models/Source.ts [hunk 1]
```diff
@@ -345,7 +346,7 @@ export class Source extends LoggedModel<Source> {
     let configObject: any = {
```
Comment (bleonard) [inline core/src/models/Source.ts:349]: Just above here it did `if (!appId || !name) return;`
Should that be `if (!appId || !modelId || !name) return;` ?

Comment (krishnaglick) [reply core/src/models/Source.ts:349]: I could have it fallback to `this.modelId`, or bail out. Whichever is fine with me.

File: core/__tests__/modules/configWriter.ts [hunk 2]
```diff
@@ -160,10 +161,12 @@ describe("modules/configWriter", () => {
       const model: GrouparooModel = await GrouparooModel.create({
         type: "profile",
         name: "Profiles",
-        id: "mod_profiles",
+        id: "mod_" + uuid.v4(),
```
Comment (krishnaglick) [inline core/__tests__/modules/configWriter.ts:164]: This simulates config mode.

File: core/__tests__/modules/configWriter.ts [hunk 3]
```diff
@@ -1033,6 +1046,7 @@ describe("modules/configWriter", () => {
           [bootstrapProperty.getConfigId()]: [12],
         },
       });
+      expect(record.modelId).toBe("mod_profiles");
```
Comment (bleonard) [inline core/__tests__/modules/configWriter.ts]: I'm suprised this isn't just `"profiles"`- unless we are in another block?
IIn that case, should we do the `uuid.v4()` thing in that block too to verify they are using the `getConfigId`?

File: core/src/models/Group.ts [hunk 4]
```diff
@@ -682,7 +684,7 @@ export class Group extends LoggedModel<Group> {
     for (const rule of convenientRules) {
       const property = await Property.findOneWithCache(
         rule.key,
-        this.modelId,
+        this.model?.getConfigId(),
```
Comment (bleonard) [inline core/src/models/Group.ts]: just `modelId` (using the variable above)

File: core/src/models/Group.ts [hunk 5]
```diff
@@ -695,7 +697,9 @@ export class Group extends LoggedModel<Group> {
       });
     }
 
-    if (!name) return;
+    if (!name || !modelId) {
```
Comment (bleonard) [inline core/src/models/Group.ts]: maybe move this up over the `const property` situation to save that work if there is `modelId`

**Review by bleonard — APPROVED** [review]
Few changes to be clearer and some test questions, but overall good to go.

File: core/__tests__/modules/configWriter.ts (hunk 3 above)
Comment (krishnaglick) [reply core/__tests__/modules/configWriter.ts]: It doesn't seem to be in another block. `expect(record.model.id).toBe("mod_profiles");` is green.
I'm not entirely certain why this happens yet.

**Review by bleonard — APPROVED** [review]

**Review by evantahler — APPROVED** [review]

**Pull request merged**